src/status_im2/contexts/wallet/common/sheets/network_preferences/view.cljs

```diff
@@ -1,70 +1,92 @@
-  [account-color]
-  [{:title        "Mainnet"
-    :image        :icon-avatar
-    :image-props  {:icon (resources/get-network :ethereum)
-                   :size :size-20}
-    :action       :selector
-    :action-props {:type                :checkbox
-                   :customization-color account-color}}])
 
-(defn- networks-list
-  [account-color]
-  [{:title        "Optimism"
-    :image        :icon-avatar
-    :image-props  {:icon (resources/get-network :optimism)
-                   :size :size-20}
-    :action       :selector
-    :action-props {:type                :checkbox
-                   :customization-color account-color}}
-   {:title        "Arbitrum"
-    :image        :icon-avatar
-    :image-props  {:icon (resources/get-network :arbitrum)
-                   :size :size-20}
-    :action       :selector
-    :action-props {:type                :checkbox
-                   :customization-color account-color}}])
+(defn- make-network-item
+  [{:keys [network-name] :as _network}
+   {:keys [color on-change network-preferences] :as _options}]
+  {:title        (string/capitalize (name network-name))
+   :image        :icon-avatar
+   :image-props  {:icon (resources/get-network network-name)
+                  :size :size-20}
+   :action       :selector
+   :action-props {:type                :checkbox
+                  :customization-color color
+                  :checked?            (contains? network-preferences network-name)
+                  :on-change           on-change}})
 
 (defn- view-internal
-  [{:keys [on-save theme]}]
-  (let [{:keys [color address]} (rf/sub [:wallet/current-viewing-account])]
-    [:<>
-     [quo/drawer-top
-      {:title       (i18n/label :t/network-preferences)
-       :description (i18n/label :t/network-preferences-desc)}]
-     [quo/data-item
-      {:status          :default
-       :size            :default
-       :description     :default
-       :label           :none
-       :blur?           false
-       :card?           true
-       :title           (i18n/label :t/address)
-       :subtitle        address
-       :container-style (merge style/data-item
-                               {:background-color (colors/theme-colors colors/neutral-2_5
-                                                                       colors/neutral-90
-                                                                       theme)})}]
-     [quo/category
-      {:list-type :settings
-       :data      (mainnet color)}]
-     [quo/category
-      {:list-type :settings
-       :label     (i18n/label :t/layer-2)
-       :data      (networks-list color)}]
-     [quo/bottom-actions
-      {:button-one-label (i18n/label :t/update)
-       :button-one-props {:disabled?           true
-                          :on-press            on-save
-                          :customization-color color}}]]))
+  []
+  (let [{:keys [color address network-preference-names]} (rf/sub [:wallet/current-viewing-account])
```

Comment:
Will leave here a link to [official doc](https://github.com/reagent-project/reagent/blob/master/doc/CreatingReagentComponents.md#form-2--a-function-returning-a-function) explaining why.

Here is an adjusted excerpt from it:
- `view-internal` is called once per component instance (and will create the state for that instance)
- the render function it returns will potentially be called many, many times. In fact, it will be called each time Reagent detects a possible difference in that component's inputs.

Reply:
@ulisesmac 
I use this subscription for initialization state in atom:
`network-preference-names-state                   (reagent/atom network-preference-names)`
Is this relevant that changes of another props - color and address should trigger re rendering this sheet component?
It's not "long-living" screen and I cannot change these params from it.

Or how can this path be improved with the initialization of the initial state?